src/ussd_optout_rapidpro.js

```diff
@@ -1,29 +1,333 @@
 go.app = function() {
     var vumigo = require("vumigo_v02");
+    var _ = require("lodash");
+    var utils = require("seed-jsbox-utils").utils;
     var App = vumigo.App;
-    var ChoiceState = vumigo.states.ChoiceState;
+    var MenuState = vumigo.states.MenuState;
     var Choice = vumigo.states.Choice;
+    var JsonApi = vumigo.http.api.JsonApi;
+    var EndState = vumigo.states.EndState;
+    var PaginatedChoiceState = vumigo.states.PaginatedChoiceState;
+    var ChoiceState = vumigo.states.ChoiceState;
 
     var GoNDOH = App.extend(function(self) {
         App.call(self, "state_start");
         var $ = self.$;
 
         self.init = function() {
+            self.rapidpro = new go.RapidPro(
+                new JsonApi(self.im, {headers: {'User-Agent': ["Jsbox/NDoH-Optout"]}}),
+                self.im.config.services.rapidpro.base_url,
+                self.im.config.services.rapidpro.token
+            );
         };
 
-        self.states.add("state_start", function(name) {
-            return new ChoiceState(name, {
-                question: $("Please let us know why you do not want MomConnect messages"),
+        self.contact_in_group = function(contact, groups){
+            var contact_groupids = _.map(_.get(contact, "groups", []), "uuid");
+            return _.intersection(contact_groupids, groups).length > 0;
+        };
+
+        self.add = function(name, creator) {
+            self.states.add(name, function(name, opts) {
+                if (self.im.msg.session_event !== 'new')
+                    return creator(name, opts);
+
+                var timeout_opts = opts || {};
+                timeout_opts.name = name;
+                return self.states.create('state_timed_out', timeout_opts);
+            });
+        };
+
+        self.states.add('state_timed_out', function(name, creator_opts) {
+            return new MenuState(name, {
+                question: $('Welcome back. Please select an option:'),
+                choices: [
+                    new Choice(creator_opts.name, $('Continue signing up for messages')),
```

Comment:
This doesn't really make sense, they're not signing up for messages on this line.

Reply:
thanks for picking this up, forgot to change the line

Reply:
thanks for picking this up